syntax = "proto3";

package org.couchers.api.constant_data;

import "google/protobuf/empty.proto";

service ConstantData {
  // This is a secure service: a user needs to be authenticated and not jailed to call functions here, refer to auth.proto and jail.proto

  rpc GetLanguageList(google.protobuf.Empty) returns (GetLanguageListRes) {
    // Gets a list of allowed language codes and their English names
  }
}

message Language {
  // ISO639-3 language code
  string code = 1;
  string name = 2;
}

message GetLanguageListRes {
  repeated Language languages = 1;
}
